docs/node/workflows.md

```diff
@@ -65,54 +65,81 @@ There is no official support for querying the state of asynchronously started Wo
+  - Equivalent to `new CancellationScope({ cancellable: true, timeout: timeoutMs }).run(fn)`.
+
+Cancellation propagates from outer scopes to inner ones and is handled by catching `CancelledError`s thrown by cancellable operations (see below).
+
+Scopes all return native JS Promises, so you can use all the familiar Promise APIs like `Promise.all` and `Promise.race` to model your async logic.
+Other APIs you can use:
+
+- `CancellationScope.current()`: get the current scope
+- `scope.cancel()`: cancel all operations inside a `scope`
+- `scope.run()`: run an async function within a `scope`
+- `scope.cancelRequested()`: a promise that resolves when a cancel request hits a `nonCancellable` workflow
 
 When a `CancellationScope` is cancelled, it propagates cancellation in any child scopes and of any cancellable operations created within it, such as:
 
 - Activities
 - Timers (created with the [`sleep`](https://nodejs.temporal.io/api/modules/workflow#sleep) function)
 - [`Trigger`](https://nodejs.temporal.io/api/classes/workflow.trigger)s
 
-## Examples
-
-### Cancel a timer from Workflow code
+### Internal Cancellation example
 
 <!--SNIPSTART nodejs-cancel-a-timer-from-workflow-->
 <!--SNIPEND-->
 
-### Alternatively, the preceding can be written as
+Alternatively, the preceding can be written as:
 
 <!--SNIPSTART nodejs-cancel-a-timer-from-workflow-alternative-impl-->
 <!--SNIPEND-->
 
-### Run multiple activities with a single deadline
+### External Cancellation example
 
-<!--SNIPSTART nodejs-multiple-activities-single-timeout-workflow-->
+Handle Workflow cancellation by an external client while an Activity is running:
+
+<!--SNIPSTART nodejs-handle-external-workflow-cancellation-while-activity-running-->
 <!--SNIPEND-->
 
-### `nonCancellable` prevents cancellation from propagating to children
+### `nonCancellable` example
+
+`CancellationScope.nonCancellable` prevents cancellation from propagating to children:
 
 <!--SNIPSTART nodejs-non-cancellable-shields-children-->
 <!--SNIPEND-->
 
-### `cancelRequested` may be awaited upon to make Workflow aware of cancellation while waiting on `nonCancellable` scopes
+### `withTimeout` example
+
+A very common operation is to cancel one or more activities if a deadline elapses:
+
+<!--SNIPSTART nodejs-multiple-activities-single-timeout-workflow-->
+<!--SNIPEND-->
+
+### `scope.cancelRequested`
+
+You can await `cancelRequested` to make Workflow aware of cancellation while waiting on `nonCancellable` scopes:
 
 <!--SNIPSTART nodejs-cancel-requested-with-non-cancellable-->
 <!--SNIPEND-->
 
-### Handle Workflow cancellation by an external client while an Activity is running
+Callbacks are not particularly useful in Workflows because all meaningful asynchronous operations return Promises.
+In the rare case that user code utilizes callbacks, `CancellationScope.cancelRequested` can be used to subscribe to cancellation.
```

Comment:
I don't understand this comment. `cancelRequested` does return `Promise`, so there is no specific requirement to consume this promise through a callback.

Reply:
Yeah, this sentence (which I wrote) isn't clear.

```suggestion
In the rare case that user code utilizes callbacks and needs to handle cancellation, a callback can be used to consume the `CancellationScope.cancelRequested` `Promise`.
```

Reply:
Maybe that's a bit better.